ablator/analysis/plot/main.py

```diff
@@ -303,21 +318,25 @@ def make_figures(
         **plt_kwargs: ty.Any,
     ):
         """
-        The function to generate violinplots for categorical values and linear plots for numerical values.
+        Generate violin plots for categorical values and linear plots for numerical values.
         Plots are created as metrics vs. attributes.
 
         Parameters
         ----------
         metric_name_remap : dict[str, str] | None
-            mappings for config's metrics keys to user defined names. By default, None
+            mappings for config's metrics keys to user defined names, by default ``None``.
         attribute_name_remap : dict[str, str] | None
-            mappings for config's searchspace names to user defined names for attributes. By default, None
+            mappings for config's searchspace names to user defined names for attributes, by default ``None``.
         save_dir : str | Path | None
             optional directory of where to save the results, when unspecified, it expects one set during
-            class initialization. By default, None
-
+            class initialization, by default ``None``.
         **plt_kwargs : ty.Any
```

Comment:
Similarly, is this a standard way to document this? Can we find some concrete examples from large libraries? e.g. matplotlib, numpy, seaborn and pandas have similar behavior where you pass arguments to nested object.

Reply:
Got it. I will review other libraries before deciding whether to use flake8 `noqa`

Reply:
Numpy docs does not even clarify the keyword arguments, they added a link to another page for these keywords arguments instead: https://numpy.org/devdocs/reference/generated/numpy.ufunc.html#:~:text=will%20remain%20uninitialized.-,**kwargs,-For%20other%20keyword

Reply:
Interestingly matplotlib does this differently, they directly display a table for the arguments: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.plot.html#:~:text=to%20autoscale_view.-,**kwargs,-Line2D%20properties%2C%20optional